src/modal.js

```diff
@@ -0,0 +1,91 @@
+const modal = document.querySelector('.modal');
+const main = document.querySelector('#main');
+
+export default function loadModal(meals, i) {
+  main.classList.add('isblur');
+  modal.classList.remove('hidden');
+  main.classList.add('noscroll');
+  modal.id = meals[i].idMeal;
```

Comment:
the i element is not the right order. Even it displays the right content, we can not consider it as a right order because of when we load data, we skip object which don't has meal name or meal photo link. So your code should use the right order which could deal with all cases not only for this specif case where the first item fulfilled the condition to have required property.